UofT - DSI | Python Assignment-2

fbastos231 opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)

- Part 1: Reading and Displaying Data from the First File
    Initiated all_paths list with all of file paths provided. Used .readlines() method and for loop to verify contents of files.

- Part 2: Data Summarization Function: 'patient_summary'
    Used .mean(), .max() and .min() methods to create patient summaries for each row of file path provided

- Part 3: Error Detection in Patient Data
    Nested patient_summary() within check_zeros() function in order to detect any errors in data

## What did you learn from the changes you have made?

for this assignment I learned how to implement .readlines(), numpy.loadtxt() methods and got more practice in creating and nesting functions as well as more about numpy.mean(), numpy.max(), numpy.min() applications in arrays. 

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?

In my opinion, the approach suggested to solve the questions were succinct and elegant. 

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?

initially attempted to use the .readlines() method to parse data into the patient_summary function. This was more complicated as this method returns 'str' type data which does not work with the numpy methods. the numpy.loadtxt() method was much better suited for this. 

## How were these changes tested?

I ran the code locally with test cells provided. Also added a couple of tests for my own understanding.

## A reference to a related issue in your repository (if applicable)

## Checklist
- [x] I can confirm that my changes are working as intended

Commit:
- Part 1: Building the base Anagram Checker
    added solution to anagram_checker. Code converts input strings into lower case, checks length
    of strings, iterates through them and if it finds matching characters adds them to a
    "matching list". The matching list length determines if anagram or not.

- Part 2: Expanding the functionality of the Anagram Checker
    added functionality to function created in part 1. New functio is checking for case
    sensitivity input, if case sensitive do not convert strings into lower case.

Commit:
- Python Assignment-2 Submission
- Part 1: Reading and Displaying Data from the First File
    Initiated all_paths list with all of file paths provided. Used .readlines() method and for loop to verify contents of files.

- Part 2: Data Summarization Function: 'patient_summary'
    Used .mean(), .max() and .min() methods to create patient summaries for each row of file path provided

- Part 3: Error Detection in Patient Data
    Nested patient_summary() within check_zeros() function in order to detect any errors in data

pedram-aliniaye-asli reviewed (approved):
You did a great job here!
Just a few quick notes:
- Please add comments to your code to make it clearer for other people and explain what is happening in your code.
- Some of the files should return `True` as the result of `detect_problems` function. But for this assignment the first file was important. You can go ahead and find the error.
Please go ahead and do the modifications and then close this pull request. Thanks!

Commit:
adding more comments for clarification of code and fixing output of 'detect_problems' function in assignment 2

fbastos231 commented:
Hi @pedram-aliniaye-asli, I added more comments to the code to clarify it. Also, the issue was that I had old output saved into the file. I had already fixed the issue with the 'detect_problems' (It was always retuning false) prior to submission but never updated the output.

Thank you!
